cmd/quadlet/main.go

```diff
@@ -210,6 +211,71 @@ func enableServiceFile(outputPath string, service *parser.UnitFile) {
 	}
 }
 
+func isImageID(imageName string) bool {
+	// All sha25:... names are assumed by podman to be fully specified
+	if strings.HasPrefix(imageName, "sha256:") {
+		return true
+	}
+
+	// However, podman also accepts image ids as pure hex strings,
+	// but only those of length 64 are unambigous image ids
+	if len(imageName) != 64 {
+		return false
+	}
+
+	for _, c := range imageName {
+		if !unicode.Is(unicode.Hex_Digit, c) {
+			return false
+		}
+	}
+
+	return true
+}
+
+func isUnambiguousName(imageName string) bool {
+	// Fully specified image ids are unambigous
+	if isImageID(imageName) {
+		return true
+	}
+
+	// Otherwise we require a fully qualified name
+	firstSlash := strings.Index(imageName, "/")
+	if firstSlash == -1 {
+		// No domain or path, not fully qualified
+		return false
+	}
+
+	// What is before the first slash can be a domain or a path
+	domain := imageName[:firstSlash]
+
+	// If its a a domain (has dot or port or is "localhost") it is considered fq
+	if strings.ContainsAny(domain, ".:") || domain == "localhost" {
+		return true
+	}
+
+	return false
+}
+
+// warns if input is an ambigious name, i.e. a partial image id or a short
+// name (i.e. is missing a registry)
+//
+// Examples:
+//   - short names: "image:tag", "library/fedora"
+//   - fully qualified names: "quay.io/image", "localhost/image:tag",
+//     "server.org:5000/lib/image", "sha256:..."
```

Comment:
```suggestion
//   - fully qualified names: "quay.io/image", "localhost/image:tag",
//     "server.org:5000/lib/image",
//   - Image IDs: "sha256:...", or 64 hex digits
```
(image IDs are not a `reference.Named`)